### PR TITLE
HRQB 17 - Prep for functional tasks and pipelines

### PR DESCRIPTION
**Pull request by ghukill**
### Purpose and background context

While beginning on writing actual tasks and pipelines to load data into Quickbase, some additional helpers and functionality were required.  This PR breaks that out into a smaller, standalone PR before implementing actual tasks.

To reiterate: this work lays the foundation for "real" tasks, which themselves should focus exclusively on getting data, transforming it, and loading it.  Anytime we can offload that work to shared functionality, the better.

The primary are fairly well encapsulated in commits for review:

- [provide convenience method to count records processed by a task](https://github.com/MITLibraries/hrqb-client/pull/22/commits/947837bf2a9580a908f54051edd7dc6bff09c901)
- [when `HRQBPipelineTask` calls another, append all pipelines to the target output path](https://github.com/MITLibraries/hrqb-client/pull/22/commits/ca67f2485e8a80135503257c59ed61b098b1fae4)
- [parameterize number of concurrent Luigi workers as env var](https://github.com/MITLibraries/hrqb-client/pull/22/commits/a5e086704897672261255a416ab40590d71a4b97)
- [`QBClient` is able to query tables for records as data used in other tasks](https://github.com/MITLibraries/hrqb-client/pull/22/commits/7d41fd2b576acacce7d7331b38db4dc7178d1e13)
- [`QBClient` tasks are considered failed if API response shows errors](https://github.com/MITLibraries/hrqb-client/pull/22/commits/04d179630e22864bda8fa0b0768a25f25894b313)
- [`QBClient` base API URL is parameterized as env var](https://github.com/MITLibraries/hrqb-client/pull/22/commits/41788deb0e6aabefe15bdc61b8fded251cda01c6)

### How can a reviewer manually see the effects of these changes?

This PR is a collection of small updates to support the writing of funtional tasks as a next step.  As such, not readily testable in their own right.

Commits have, where applicable, new tests to demonstrate the new behavior.  

### Includes new or updated dependencies?
NOI

### Changes expectations for external applications?
NO

### What are the relevant tickets?
- https://mitlibraries.atlassian.net/browse/HRQB-17

### Developer
- [X] All new ENV is documented in README
- [X] All new ENV has been added to staging and production environments
- [X] All related Jira tickets are linked in commit message(s)
- [X] Stakeholder approval has been confirmed (or is not needed)

### Code Reviewer(s)
- [ ] The commit message is clear and follows our guidelines (not just this PR message)
- [ ] There are appropriate tests covering any new functionality
- [ ] The provided documentation is sufficient for understanding any new functionality introduced
- [ ] Any manual tests have been performed **or** provided examples verified
- [ ] New dependencies are appropriate or there were no changes

### CODE REVIEW

File: hrqb/utils/quickbase.py [hunk 1]
```diff
@@ -50,15 +51,25 @@ def make_request(
...
-        results = requests_method(
+        response = requests_method(
             f"{self.api_base}/{path.removeprefix('/')}",
             headers=self.request_headers,
             **kwargs,
-        ).json()
+        )
+
+        # handle non 2xx responses
+        if not 200 <= response.status_code < 300:  # noqa: PLR2004
```
Comment (ehanson8) [inline hrqb/utils/quickbase.py:61]: Curious why filtering below 300?

**Review by ehanson8 — APPROVED** [review]
Logic seems sound, 1 non-blocking question

File: hrqb/utils/quickbase.py (hunk 1 above)
Comment (ghukill) [reply hrqb/utils/quickbase.py:61]: Generally speaking, any `2xx` status codes, literally less than `300`, are considered successful requests.  While the response may further suggest if something unusual happened, the request/response was successful.

I didn't see any built-in ways with the `requests` library to look for `2xx` messages.  They do have some built-in Enums like `OK`, but that resolves to only `200`.  Example: some QB API responses are `207`, which is successful, but that's not defined anywhere in `requests` as such.

So, any `2xx` status code, quite literally `< 300` seems sufficient.

Comment (ehanson8) [reply hrqb/utils/quickbase.py:61]: Got it, thanks for the context!

**Review by jonavellecuerdo — APPROVED** [review]
Looks good to me!

**Pull request merged**